lib/config_validator.rb

```diff
@@ -1,22 +1,37 @@
 class ConfigValidator
+  ENV_PREFIX = Figaro::Application::FIGARO_ENV_PREFIX
+
   def validate(env = ENV)
     bad_keys = keys_with_bad_values(env, candidate_keys(env))
     return unless bad_keys.any?
-    raise warning(bad_keys).tr("\\\n", ' ')
+    raise warning(bad_keys).tr("\n", ' ')
   end
 
   private
 
   def candidate_keys(env)
-    env.keys.delete_if { |key| key.starts_with?(Figaro::Application::FIGARO_ENV_PREFIX) }
+    env.keys.keep_if { |key| candidate_key?(env, key) }
+  end
+
+  def candidate_key?(keys, key)
+    # A key is associated with a configuration setting if there are two
+    # settings in the environment: one with and without the Figaro prefix.
+    # We're only interested in the configuration settings and not other
+    # environment variables.
+
+    keys.include?(key) and keys.include?(ENV_PREFIX + key)
   end
 
   def keys_with_bad_values(env, keys)
-    keys.keep_if { |key| %w[yes no].include?(env[key]) }
+    # Configuration settings for boolean values need to be "true/false"
+    # and not "yes/no".
+
+    keys.keep_if { |key| env[key].strip.downcase =~ /^(?:yes|no)$/ }
```

Comment:
Was there a particular reason for switching to regex? It's almost twice as slow as checking the array. Can we do this instead?
```ruby
%w[yes no].include?(env[key].strip.downcase)
```